Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/filter/typefilters/CaseSectionEventAttributeFilter.java

```diff
@@ -44,30 +44,37 @@ public static boolean toKeep(ATrace trace, LogFilterRule logFilterRule) {
 
     private static boolean conformRule(ATrace trace, LogFilterRule logFilterRule) {
         String attributeKey = logFilterRule.getKey();
-        Set<String> values = logFilterRule.getPrimaryValuesInString();
+
+        Set<String> values = (Set<String>) logFilterRule.getPrimaryValues().iterator().next().getObjectVal();
```

Comment:
Guaranteed logFilterRule.getPrimaryValues() can never be empty?  Judging by the conditional on line 51, this doesn't seem to be the case.